### PR TITLE
Remove hard file offset reset in load()

### PR DESCRIPTION
**Pull request by philippslang**
Slight improvement in file offset handling to be consistent with save() and pickle API, enables loading from arbitrary offsets.

#3694

### CODE REVIEW

File: torch/serialization.py [hunk 1]
```diff
@@ -389,13 +389,20 @@ def persistent_load(saved_id):
         else:
             raise RuntimeError("Unknown saved id type: %s" % saved_id[0])
 
-    # try the legacy loader first, which only works if f is a tarfile
+    # on windows this can return invalid values
```
Comment (colesbury) [inline torch/serialization.py]: What kind of invalid values? When does it return invalid values?

Comment (plang85) [reply torch/serialization.py]: right, I see. we know we're in binary mode so this point is mute. it's for unix file endings in text mode that this applies

**Review by colesbury — APPROVED** [review]
lgtm

**Comment by soumith** [comment]
@apaszke didn't you want to comment on this PR, you said something was missing?

**Review by apaszke — APPROVED** [review]

**Comment by soumith** [comment]
thank you  @plang85 !

**Pull request merged**